Podcast/CardTableViewCellContextView.swift

```diff
@@ -0,0 +1,159 @@
+        contextImages.spacing = -1 * contextImagesSize
+        addSubview(contextImages)
+        
+        contextImages.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(contextMarginX)
+            make.height.equalTo(contextImagesSize)
+            make.centerY.equalToSuperview()
+        }
+        
+        contextLabel.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.leading.equalTo(contextImages.snp.trailing).offset(marginSpacing)
+            make.trailing.equalToSuperview().inset(contextLabelRightX)
+        }
+        
+        feedControlButton.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.width.equalTo(feedControlButtonWidth)
+            make.height.equalTo(feedControlButtonHieght)
+            make.trailing.equalTo(feedControlButtonRightX)
+        }
+        
+        topLineSeperator.snp.makeConstraints { make in
+            make.bottom.equalToSuperview()
+            make.trailing.equalToSuperview()
+            make.leading.equalToSuperview()
+            make.height.equalTo(lineSeperatorHeight)
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func prepareForReuse() {
+        for view in contextImages.subviews {
+            view.removeFromSuperview()
+        }
+    }
+    
+    func setupWithCard(episodeCard: EpisodeCard) {
+        if let recommendCard = episodeCard as? RecommendedCard {
+            setRecommendedCard(card: recommendCard)
+        } else if let releaseCard = episodeCard as? ReleaseCard {
+            setReleaseCard(card: releaseCard)
+        } else if let tagCard = episodeCard as? TagCard {
+            setTagCard(card: tagCard)
+        }
+    }
+    
+    internal func setRecommendedCard(card: RecommendedCard) {
+        if card.namesOfRecommenders != [] {
+            contextLabel.text = ""
+            for i in 0..<card.namesOfRecommenders.count {
+                contextLabel.text = contextLabel.text! + card.namesOfRecommenders[i]
+                if i != card.namesOfRecommenders.count - 1 {
+                    contextLabel.text = contextLabel.text! + ", "
+                }
+            }
+            if card.numberOfRecommenders > 3 {
+                contextLabel.text = contextLabel.text! + ", and " + String(card.numberOfRecommenders - 3) + " others recommended this podcast"
+            } else {
+                contextLabel.text = contextLabel.text! + " recommended this podcast"
+            }
+        }
+        
+        for i in 0..<card.imageURLsOfRecommenders.count {
+            let imageView = ImageView(frame: CGRect(x: 0, y: 0, width: contextImagesSize, height: contextImagesSize))
+            contextImages.addArrangedSubview(imageView)
+            layoutContextImageView(imageView: imageView, imageURL: card.imageURLsOfRecommenders[i])
+        }
```

Comment:
For each is most appropriate here rather than indexing
for url in card.imageURLsOfRecommenders {

Reply:
Done